src/screens/settings-screen.tsx

```diff
@@ -1,63 +1,208 @@
+import {
+	deleteTileCacheDirectory,
+	makeTileCacheDirectory,
+} from "../utils/file-system";
+import { SettingsScreen, SettingsData } from "react-native-settings-screen";
 import SettingsToggle from "../components/settings-toggle";
-import theme from "../theme";
+import ModalSelector from "react-native-modal-selector";
 
-/** Primitive Settings Screen implementation
- * Light colored separator and hairline ruler to separate content
- * Primitive layout with flex
- */
+export const SettingScreen = () => {
+	const [userId, currTrack, currSend] = useTypedSelector((state) => [
+		state.user.userId,
+		state.user.trackingInterval / 1000,
+		state.user.sendingInterval / 1000,
+	]);
 
-const Ruler = () => <View style={styles.rulerStyle} />;
-const Separator = () => <View style={styles.separatorStyle} />;
+	const dispatch = useTypedDispatch();
+
+	const alertOnReset = (target: string) => {
+		Alert.alert(
+			`Resetting ${target}`,
+			`Do you really want to reset ${target}?`,
+			[
+				{
+					text: "Cancel",
+				},
+				{
+					text: "Reset",
+					onPress: async () => {
+						if (target === "UserID") {
+							await dispatch(resetUser());
+							await dispatch(identifyUser());
+						}
+						if (target === "maptile cache") {
+							deleteTileCacheDirectory();
+							makeTileCacheDirectory();
+						}
+					},
+				},
+			]
+		);
+	};
+
+	let index = 0;
+	const trackFreq = [
+		{ key: index++, component: <Text>1 second</Text>, label: 1000 },
+		{ key: index++, component: <Text>5 seconds</Text>, label: 5000 },
+		{ key: index++, component: <Text>10 seconds</Text>, label: 10000 },
+		{ key: index++, component: <Text>30 seconds</Text>, label: 30000 },
+		{ key: index++, component: <Text>1 minute</Text>, label: 60000 },
+	];
+
+	index = 0;
+	const sendFreq = [
+		{ key: index++, component: <Text>10 seconds</Text>, label: 10000 },
+		{ key: index++, component: <Text>30 seconds</Text>, label: 30000 },
+		{ key: index++, component: <Text>1 minute</Text>, label: 60000 },
+		{ key: index++, component: <Text>5 minutes</Text>, label: 300000 },
+		{ key: index++, component: <Text>10 minutes</Text>, label: 600000 },
+	];
+
+	index = 0;
+	const tileLifetime = [
+		{ key: index++, component: <Text>12 hours</Text>, label: 43200000 },
+		{ key: index++, component: <Text>24 hours</Text>, label: 86400000 },
+		{ key: index++, component: <Text>48 hours</Text>, label: 172800000 },
+	];
+
+	const settingsData: SettingsData = [
+		{
+			type: "SECTION",
+			header: "Navigation".toUpperCase(),
+			footer:
+				"Change tracking and sending frequencies to save battery life and turn the app to offline mode",
+			rows: [
+				{
+					title: "Waypoint tracking frequency",
+					renderAccessory: () => (
+						<ModalSelector
+							data={trackFreq}
+							initValue={currTrack.toString()}
+							onChange={async (option) => {
+								await dispatch(setInterval(option.label, true));
+							}}
```

Comment:
There seems to TS problem `JSX element class does not support attributes because it does not have a 'props' property.ts(2607)` when `ModelSelectors` are rendered. Also `option.label` parameter is missing type `Parameter 'option' implicitly has an 'any' type.ts(7006)`